pkg/linters/components/compimpl/linter.go

```diff
@@ -0,0 +1,143 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package compimpl is a linter that checks if the component implementation follows the expected naming convention.
+package compimpl
+
+import (
+	"fmt"
+	"go/ast"
+	"os"
+	"regexp"
+	"strings"
+
+	"golang.org/x/tools/go/analysis"
+)
+
+func init() {
+	RegisterLinter("compimpl", "Check the component implementation follows the naming convention", run)
+}
+
+func run(pass *analysis.Pass) (interface{}, error) {
+	r := regexp.MustCompile(".*comp/.*/.*/.*impl/.*")
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏Does it behave as expected on Windows ? In particular is the file path given in unix format or windows format ?

Reply:
Code updated.